base/multi.jl

```diff
@@ -1544,7 +1544,7 @@ function pmap(f, lsts...; err_retry=true, err_stop=false, pids = workers())
     nextidx = 0
     getnextidx() = (nextidx += 1)
 
-    states = [start(lsts[idx]) for idx in 1:len]
+    states = map(start, lsts)
```

Comment:
One risk with using `map` here is that you can't guarantee that `states` is an array. For example, if `lsts` is a tuple, then `states` will be a tuple, and the later line that sets `states[i] = ...` will fail.